libp2p/src/builder.rs

```diff
@@ -113,7 +120,57 @@ mod tests {
     }
 
     #[test]
-    #[cfg(all(feature = "tokio", feature = "tcp", feature = "tls", feature = "mplex"))]
+    #[cfg(all(feature = "tokio", feature = "quic"))]
+    fn quic() {
+        let _ = SwarmBuilder::with_new_identity()
+            .with_tokio()
+            .with_quic()
+            .with_behaviour(|_| libp2p_swarm::dummy::Behaviour)
+            .unwrap()
+            .build();
+    }
+
+    #[test]
+    #[cfg(all(feature = "async-std", feature = "quic"))]
+    fn async_std_quic() {
+        let _ = SwarmBuilder::with_new_identity()
+            .with_async_std()
+            .with_quic()
+            .with_behaviour(|_| libp2p_swarm::dummy::Behaviour)
+            .unwrap()
+            .build();
+    }
+
+    #[test]
+    #[cfg(all(feature = "tokio", feature = "quic"))]
+    fn quic_config() {
+        let _ = SwarmBuilder::with_new_identity()
+            .with_tokio()
+            .with_quic_config(|config: libp2p_quic::Config| config)
```

Comment:
```suggestion
            .with_quic_config(|config| config)
```

Do we need the annotation?

Reply:
Not really. I can remove it

Reply:
done